rust/Makefile

```diff
@@ -99,7 +99,7 @@ $(objtree)/rust/bindings_generated.rs: $(srctree)/rust/kernel/bindings_helper.h
 quiet_cmd_exports = EXPORTS $@
       cmd_exports = \
 	$(NM) -p --defined-only $< \
-		| grep -E ' (T|R|D) ' | cut -d ' ' -f 3 | grep -E '^(__rust_|_R)' \
+		| grep -E ' (T|R|D) ' | cut -d ' ' -f 3 \
```

Comment:
This is to expose the `no_mangle`-ed symbol so functions can be called from C-side. I actually don't understand why there is the filtering in the first place, since `nm` is only showing defined symbols anyway.

Reply:
The ones that would appear would be:

    __rdl_oom
    __rg_oom
    __rg_alloc
    __rg_alloc_zeroed
    __rg_dealloc
    __rg_realloc
    rust_oom

Reply:
Sorry I don't quite get it, why shouldn't they appear?

Reply:
No, I was simply stating the ones for information.

I do not remember if there were other symbols being included when I wrote it originally. Perhaps I was trying to ensure we did not expose anything else, or trying to use the smaller set I could.

Should those 7 be exposed, i.e. do we expect `rustc` to emit calls to those? (even from e.g. generics?)

Reply:
I mean emitting calls to those from leaf modules or external third-party modules; and by "from generics" I mean that e.g. we could have a call in one of the generics in `kernel` and then that one get generated when generating instantiations of the generic in a leaf module etc.

Reply:
If it is not something that can be easily answered, or not guaranteed, or something like that, we can remove the filtering, it is not a big deal.

Reply:
I checked the disassembly, and it seems they are never called directly, but always through `__rust_alloc`, `__rust_alloc_error_handler` and friends. Anyway even if we want some filter it should be a denylist rather than by symbol prefixes?

Reply:
Yeah, the prefixes approach was intended to pick the "Rust-y" symbols. If we go for a list, perhaps the opposite: only allowing the ones we really need. We can put the adding the list on an issue.

Reply:
I'll just leave it as is then. We can open an issue to investigate what's best approach for symbol exports.